mreg_cli/commands/registry.py

```diff
@@ -0,0 +1,52 @@
+"""Main command wrapper interface for mreg-cli.
+
+This module provides the :py:class:`CommandWrapper` class, which is used to wrap
+a command and its attributes, so it may be used as a subcommand of mreg-cli.
+"""
+
+import argparse
+from typing import Callable, List, Optional
+
+from mreg_cli.types import Command, Flag
+
+
+class CommandRegistry:
+    """A registry of commands that can be registered with the CLI.
+
+    This class is used to register commands with the CLI. It provides a
+    decorator, :py:meth:`register_command`, which can be used to register a
+    command with the CLI.
+    """
+
+    def __init__(self) -> None:
+        """Initialize the command registry."""
+        self._commands: List[Command] = []
+
+    def register_command(
+        self,
+        prog: str,
+        description: str,
+        short_desc: str,
+        flags: Optional[List[Flag]] = None,
+    ) -> Callable[[argparse.Namespace], None]:
```

Comment:
Currently, mypy gives us the following error:

```
mreg_cli/commands/zone.py:127: error: Argument 1 has incompatible type "Callable[[Namespace], None]"; expected "Namespace"
```

The current annotation claims this method returns a command function, which is incorrect. The `register_command()` method actually returns a decorator function that takes a command function which THEN returns said command function. 


It might be useful to add an alias for command functions to `types.py`:

```py
# types.py
CommandFunc = Callable[[argparse.Namespace], None]
```

Which would allow us to do this:

```suggestion
    ) -> Callable[[CommandFunc], CommandFunc]:
```

as opposed to the more verbose:

```suggestion
    ) -> Callable[[Callable[[argparse.Namespace], None]], Callable[[argparse.Namespace], None]]:
```

Reply:
Yes, thank you! I'm disappointed that pylance didn't catch that return issue. Bleeding obvious when you look at it though. Fixed in https://github.com/terjekv/mreg-cli/commit/38032ccb70ef7e21a03ee1d2898d4abd7ff1a792 (but there are lots of other mypy issues as well, mostly inherited.)